org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -384,10 +384,19 @@ private void syncCapabilitiesToSettings() {
 	 */
 	private void synchronizeBundles() {
 		try {
-			List<String> bundlesToRefresh = (ArrayList<String>) JavaLanguageServerPlugin.getInstance()
+			Object commandResult = JavaLanguageServerPlugin.getInstance()
 				.getClientConnection().executeClientCommand("_java.reloadBundles.command");
-			if (bundlesToRefresh.size() > 0) {
-				BundleUtils.loadBundles(bundlesToRefresh);
+			if (commandResult instanceof List<?> list) {
+				List<String> bundlesToRefresh = (List<String>) commandResult;
+				if (bundlesToRefresh.size() > 0) {
+					BundleUtils.loadBundles(bundlesToRefresh);
+				}
+			} else if (commandResult instanceof Map<?, ?> m) {
```

Comment:
Not a `ResponseError` here ? It's just a Map ?

Reply:
Right, looks like because we define the return type as `Object` at https://github.com/eclipse/eclipse.jdt.ls/blob/438e6192cc327960f82d3086bc5f374a5cfaeca7/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/lsp/ExecuteCommandProposedClient.java#L23-L24 , we just get some very basic types. `@ResponseJsonAdapter(...)` would be needed to get a custom type.